### ISSUE TITLE
bug: running datetime in ios mode in chrome causes swiping to get stuck

### ISSUE THREAD

**Issue by MilotH2** [ionic-team/ionic-framework#24679]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

So, on iOS everything is OK, the issue is that on Android, everything that is a ion-datetime with Popover/Modal, it does not work, it gets stuck. Another issue that I saw during the tests before writing this, is that if the datetime is on the bottom of the page, and the datetime is a popover it shows the datetime under the content, so it is not displayed properly.

One thing i saw is that when I SWIPE it get's stuck before the end of the swipe. I saw then when i clicked the arrow on the top right for next month, it completed the swipe.

The same issue is when you go to official ionic datetime page and test on the modal datetime presentation, it get's stuck.

This ion-datetime works only on PAGE, even if i open a MODAL/POPOVER on it's own and add a normal ion-datetime the issue is still there. We need this fix ASAP since otherwise I AM NOT ABLE TO SELECT DATES AS THEY SHOULD BE SELECTED ON MODALS/POPOVERS. 

Have been testing on livereload and on production as APK.

EDIT: I JUST SAW THAT I HAD SET MODE: IOS on APP MODULE, I ALSO SAW THAT ON ANDROID THE ION-MODAL MUST BE MODE MD OTHERWISE IT WONT WORK, BUT THIS WORKS ONLY ON PAGE AND ON MODAL AS NORMAL ION-DATETIME.

EDIT: I CAN CONFIRM THAT ION-DATETIME with MODE="MD" on MODAL AS MODAL POPOVER NOW WORKS, I HAD TO SET MODE="MD" like this:

```
<ion-item style="--ripple-color: transparent" button="true" id="open-modal2" >
              <ion-label>{{dateValue2}}</ion-label>
              <ion-modal mode="md" trigger="open-modal2">
                <ng-template mode="md">

                  <ion-content mode="md" force-overscroll="false">
                    <ion-datetime
                    [min]="minimumDateTo"
                    #popoverDatetime2
                    presentation="date"
                    (ionChange)="dateValue2 = formatDate(popoverDatetime2.value);"
                    formControlName="date_till">
                      <div slot="title">{{'date_to' | translate}}</div>
                    </ion-datetime>
                  </ion-content>
                </ng-template>
              </ion-modal>
            </ion-item>
```
Otherwise, the popoverstyle still is buggy so i switched to modal style. Thanks.

https://user-images.githubusercontent.com/55019464/151992192-6f1fb4be-ae49-4f43-a9ea-0adb84e0f54e.mp4

https://user-images.githubusercontent.com/55019464/151992229-fa6b15a0-c189-493e-bd10-1f13e2b383d5.mp4

https://user-images.githubusercontent.com/55019464/151992234-54997b9b-a316-4e13-8e29-cc9c56e6090b.mp4

https://user-images.githubusercontent.com/55019464/151992239-5dfa4630-b637-4255-8f4e-f31b5382ce05.mp4

https://user-images.githubusercontent.com/55019464/151992243-26a23675-e13a-461a-b057-36b6592ccd7e.mp4

https://user-images.githubusercontent.com/55019464/151992245-01019a94-ec15-4e1f-a1f1-beaa2fc72fda.mp4

https://user-images.githubusercontent.com/55019464/151992251-16030dd0-7ff1-49d8-b089-e90b8f4c52bb.mp4

### Expected Behavior

Should work as normal as on a page, to be able to swipe between months without a problem, and to be able to select dates on Popover/Modals.

A solution of this would be to be able to disable swipe and keep only RIGHT LEFT arrows on MODALS, while on Popovers it wasn't working both ways.

Also another solution would be to use the old ion-datetime, give us an opportunity to be able to use old ion-datetime style, so that this would be just a workaround till the new ion-datetime would get fixed.

### Steps to Reproduce

On this section you can also check the screen record on official web page of ionic

ON PAGE normal ion-datetime works:
`
<ion-datetime>
  <div slot="title">My Custom Title</div>
</ion-datetime>
`

ON PAGE AS POPOVER does not work, even if you have the button of datetime on bottom of the page, it shows very bad and unable to see all the dates:

`
<ion-item button="true" id="open-date-input">
  <ion-label>Datea</ion-label>
  
  <ion-popover  trigger="open-date-input" show-backdrop="false">
    <ng-template>
      <ion-datetime
        #popoverDatetime
        presentation="date"
      ></ion-datetime>
    </ng-template>
  </ion-popover>
</ion-item>
`

ON MODAL AND ON POPOVER, copied the source code from the official ionic documentation, also copied the styles so it would apply as it should, still same behavior, only that on modal type datetime you could scroll between months by clicking the right-left arrays.

`
   <div class="grid">
              <div class="grid-item">
                <h2>Datetime in Overlay</h2>
                <ion-button id="open-modal">Open Datetime Modal</ion-button>
                <ion-modal  mode="ios" trigger="open-modal">
                  <ng-template mode="ios">
                  <ion-content mode="ios" >
                    <ion-datetime mode="ios"></ion-datetime>
                  </ion-content>
                  </ng-template>
                </ion-modal>
              </div>
            </div>
`
https://user-images.githubusercontent.com/55019464/151995273-e020b32f-74e6-4e2a-8cd5-9f98a5d2bc54.mp4

### Code Reproduction URL

https://github.com/MilotH2/blankAppForTestingIonDatetimeIusse.git

### Ionic Info

ionic info

Ionic:

   Ionic CLI                     : 6.18.1 (/Users/milothoxha/.nvm/versions/node/v16.13.1/lib/node_modules/@ionic/cli)
   Ionic Framework               : @ionic/angular 6.0.4
   @angular-devkit/build-angular : 13.0.4
   @angular-devkit/schematics    : 13.0.4
   @angular/cli                  : 13.0.4
   @ionic/angular-toolkit        : 5.0.3

Capacitor:

   Capacitor CLI      : 3.3.3
   @capacitor/android : 3.3.3
   @capacitor/core    : 3.3.3
   @capacitor/ios     : 3.3.3

Utility:

   cordova-res : 0.15.4
   native-run  : 1.5.0

System:

   NodeJS : v16.13.1 (/Users/milothoxha/.nvm/versions/node/v16.13.1/bin/node)
   npm    : 8.1.2
   OS     : macOS Big Sur

### Additional Information

There are planty of videos that describe the issue, the video here is that to check that on iOS works as it should, except the POPOVER style which it shows on the bottom of the screen and that the dates are not visible

https://user-images.githubusercontent.com/55019464/151992420-1ed40548-6422-4b40-bba4-0f658ebb9bfa.mp4

.

**Comment by sean-perkins** [comment]
@MilotH2 thanks for reporting these issues.

The swipe issue that you mention has been resolved in v6.0.5. Unfortunately in 6.0.5, I introduced an issue where the next/previous buttons would skip by 2 months. This has been also fixed, but won't be available until v6.0.6 next week.

Can you please create a reproduction app for the datetime positioning issue in a popover when it is overflowing off the bottom of the screen?

**Comment by MilotH2** [comment]
@sean-perkins it is a pleasure :).
There is already a reproduction app for the datetime popover positionin issue, check it out. 
https://github.com/MilotH2/blankAppForTestingIonDatetimeIusse.git


Also, the best case scenario that the Ion-Datetime was working for me was to use Mode=MD with Modal Style, only in this case i was not having any issues. I will share the code I am using:

Html:
```
 <ion-item style="--ripple-color: transparent" button="true" id="open-modal2" >
        <ion-label>{{dateValue2}}</ion-label>
        <ion-modal mode="md" trigger="open-modal2">
          <ng-template mode="md">

            <ion-content mode="md" force-overscroll="false">
              <ion-datetime
              [min]="minimumDate"
              #popoverDatetime2
              presentation="date"
              [max]="2050"
              (ionChange)="dateValue2 = formatDate(popoverDatetime2.value);"
              formControlName="date">
                <div slot="title">{{'delay_date' | translate}}</div>
              </ion-datetime>
            </ion-content>
          </ng-template>
        </ion-modal>
      </ion-item>
```

CSS:
```

ion-datetime {
  height: auto;
  width: auto;

  max-width: 350px;
}


ion-modal {
  --width: 290px;
  --height: 382px;
  --border-radius: 8px;
  --backdrop-opacity: 0.6 !important;
}

ion-modal ion-datetime {
  height: 382px;
}

```

TypescriptFile:
```



  @ViewChild('dataa', {static: false}) dataa: IonDatetime;
  dateValue2: any = this.translateService.instant('select_date');
 formatDate(value: string) {
    return format(parseISO(value), 'MMM dd yyyy');
  }


```

This way I had no more issues as on Page, Popover or Modal components.
Keep up the great work.

**Comment by antoine92190** [comment]
@sean-perkins I still reproduce the issue when swiping is blocked on Android device using `[mode]="'ios'"`

**Comment by MilotH2** [comment]
@antoine92190 remember, do not do mode ios on android platform, otherwise it will crash, on android keep with mode md.

**Comment by antoine92190** [comment]
@MilotH2 Yes I understand, but I would like to use `ios` mode on Android. As the option exists, it's expected to be working.

**Comment by liamdebeasi** [comment]
Hi everyone,

Just getting caught up on this, but the issue here is related to https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/datetime/datetime.tsx#L741.

This code causes `pointer-events:` to get disabled after swiping the calendar body part of the way. This was done to work around a bug in WebKit, the engine that powers Safari and WKWebView on iOS: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/datetime/datetime.tsx#L731-L736

We intended for the iOS mode `ion-datetime` to be used only on iOS devices. However, when running the same code in a non-WebKit browser, you will see the swipe gesture get stuck, as noted in this thread.

As a workaround, we recommend setting the datetime mode to `md` when running on an Android device.

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved in the latest version of Ionic.
